coordinate-distance-calculator-app/pages/index.js

```diff
@@ -0,0 +1,84 @@
+import { useState } from "react";
+import axios from "axios";
+
+export default function Home() {
+  const [firstCoordinateSet, setFirstCoordinateSet] = useState("");
+  const [secondCoordinateSet, setSecondCoordinateSet] = useState("");
+  const [res, setRes] = useState("");
+  const [btnText, setBtnText] = useState("Calculate");
+
+  /**
+   *
+   *
+   *
+   */
+  const fetchCoordinatesDistance = async (e) => {
+    e.preventDefault();
+
+    try {
+      setBtnText("Calculating...");
+      const res = await axios.get(`/api/calculator`, {
+        params: {
+          firstCoordinateSet,
+          secondCoordinateSet,
+        },
+      });
+      setRes(res.data);
+    } catch (err) {
+      console.log(err);
+    }
+    setBtnText("Calculate");
+  };
+
+  return (
+    <div className="flex flex-col items-center relative min-h-screen">
+      <h2 className="font-raleway font-bold text-center text-6xl text-primary pt-20 pb-6 md:text-3xl">
+        Coordinate <span className="text-secondary">Distance</span> Calculator
+        App
+      </h2>
+      <h3 className="text-lightGrey text-2xl font-raleway font-bold uppercase tracking-wide mb-12 md:text-base md:px-4 md:text-center">
+        Calculate the distance between two coordinates
+      </h3>
+      <div className="flex flex-col justify-between items-center w-full md:items-center">
+        <form className="flex w-full justify-center md:flex-col md:w-5/6">
+          <input
+            autoFocus={true}
+            className="border-none outline-none bg-primary px-4 py-2 w-1/6 mx-2 rounded-sm font-raleway md:w-full md:mx-0"
+            placeholder="Enter first coordinate set..."
+            onChange={(e) => setFirstCoordinateSet(e.target.value)}
```

Comment:
It would be good to pre-fill ta input fields with sample coordinates. Also, it's a better UX to show an error message when the request fails.